app.rb

```diff
@@ -0,0 +1,170 @@
+require_relative './person'
+require_relative './student'
+require_relative './book'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  def initialize
+    @books = Book.class_variable_get(:@@books)
+    @people = Person.class_variable_get(:@@people)
+    @rentals = Rental.class_variable_get(:@@rentals)
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    loop do
+      puts "
+          1 | List all books
+          2 | List all people
+          3 | Create a person
+          4 | Create a book
+          5 | Create a rental
+          6 | List all rentals for a given person id
+          7 | Exit"
+      puts "\nSelect an option: "
+      user_input = gets.chomp.to_i
+      break if user_input >= 7
+
+      puts "\n"
+      options user_input
+    end
+  end
+
+  def options(option)
+    case option
+    when 1
+      list_books
+    when 2
+      list_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    else
+      puts 'Unknown option'
+    end
+  end
+
+  def list_books
+    puts '*' * 70
+    if @books.length.zero?
+      (puts 'Book list is empty')
+    else
+      puts 'Book list:'
+      @books.each_with_index do |book, index|
+        puts "#{index + 1} | #{book.title} - By #{book.author}"
+      end
+    end
+    puts '*' * 70
+  end
+
+  def list_people
+    puts '*' * 70
+    if @people.length.zero?
+      (puts 'People list is empty')
+    else
+      puts 'People list:'
+      @people.each_with_index do |person, index|
+        puts "[Student] #{index + 1} | Name: #{person.name} - Age: #{person.age}" if person.is_a?(Student)
+        puts "[Teacher] #{index + 1} | Name: #{person.name} - Age: #{person.age}" if person.is_a?(Teacher)
+      end
```

Comment:
- [ ] Let's also make sure we print the person's `id` so that it can be used to list all rentals for a given person `id`.